services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/hakemus/HakemusControllerITest.kt

```diff
@@ -687,6 +690,128 @@ class HakemusControllerITest(@Autowired override val mockMvc: MockMvc) : Control
         }
     }
 
+    @Nested
+    inner class OperationalCondition {
+        private val url = "/hakemukset/$id/toiminnallinen-kunto"
+        private val date = LocalDate.of(2024, 8, 8)
+        private val request = DateReportRequest(date)
+
+        @Test
+        @WithAnonymousUser
+        fun `returns 401 when unknown user`() {
+            post(url, request).andExpect(status().isUnauthorized)
+
+            verifySequence { hakemusService wasNot Called }
+        }
+
+        @Test
+        fun `returns 400 when no request body`() {
+            post(url).andExpect(status().isBadRequest)
+
+            verifySequence { hakemusService wasNot Called }
+        }
+
+        @Test
+        fun `returns 200 when date is sent successfully`() {
+            every {
+                authorizer.authorizeHakemusId(id, PermissionCode.EDIT_APPLICATIONS.name)
+            } returns true
+            justRun { hakemusService.operationalCondition(id, date) }
+
+            post(url, request).andExpect(status().isOk).andExpect(content().string(""))
+
+            verifySequence {
+                authorizer.authorizeHakemusId(id, PermissionCode.EDIT_APPLICATIONS.name)
+                hakemusService.operationalCondition(id, date)
+            }
+        }
+
+        @Test
+        fun `returns 404 when the application is not found`() {
+            every {
+                authorizer.authorizeHakemusId(id, PermissionCode.EDIT_APPLICATIONS.name)
+            } throws HakemusNotFoundException(id)
+
+            post(url, request)
+                .andExpect(status().isNotFound)
+                .andExpect(hankeError(HankeError.HAI2001))
+
+            verifySequence {
+                authorizer.authorizeHakemusId(id, PermissionCode.EDIT_APPLICATIONS.name)
+            }
+        }
+
+        private fun RuntimeException.whenThrown(f: () -> Unit) {
+            every {
+                authorizer.authorizeHakemusId(id, PermissionCode.EDIT_APPLICATIONS.name)
+            } returns true
+            every { hakemusService.operationalCondition(id, date) } throws this
+
+            f()
+
+            verifySequence {
+                authorizer.authorizeHakemusId(id, PermissionCode.EDIT_APPLICATIONS.name)
+                hakemusService.operationalCondition(id, date)
+            }
+        }
+
+        @Test
+        fun `returns 409 when the application is not yet in Allu`() {
+            val exception = HakemusNotYetInAlluException(HakemusFactory.create(id = id))
+
+            exception.whenThrown {
```

Comment:
For me this "whenThrown" reads in my mind that "when this exception is thrown, do the following things". Instead, the inner function call throws this exception. It is good that with this the test methods are more clean but this syntax is a bit hard to read. If it was "exception.thrownWhen" or "isThrownWhen" then perhaps its purpose was more clear?

Reply:
I was going for "When the mock service throws an exception".

`withMockedException(exception)`?

Reply:
I renamed this to `withMockedException(exception)`.